### PR TITLE
Jailbreaking iOS section edited and enhanced.

### PR DESCRIPTION
**Pull request by cpholguera**

### CODE REVIEW

**Comment by cpholguera** [comment]
please contact me first before starting to review it. There are some things to be clarified before I commit the final changes to the PR. Like for example the references.

**Comment by Sjord** [comment]
Thank you for your pull request. What do you want to know before we can start reviewing it? You mention references. We recently switched to [inline references](https://github.com/OWASP/owasp-mstg/issues/590) using markdown links.

**Comment by cpholguera** [comment]
@Sjord It's now resolved. I fixed the inline references as well. Now it's ready for review. Thanks :)

File: Document/0x06b-Basic-Security-Testing.md [hunk 1]
```diff
@@ -27,36 +27,63 @@ If you want to get serious with iOS security testing, you need a Mac, for the si
...
+- Providing root access to the operating system
+- Allowing applications and tools not signed by Apple to be installed and run without any restrictions
+- Debugging and performing dynamic analysis
+- Providing access to the Objective-C Runtime
+
+iOS applications store data in the application sandbox which is not accessible to the public (but is available to root and the application itself). Without root access, it is not possible to access the application sandbox and analyze the data that was stored in it.
+
+#### Caveats and Considerations about Jailbreaking
+
+Jailbreaking iOS devices is becoming more and more complicated as Apple keeps hardening the system, patching the corresponding vulnerabilities that jailbreaks are based on and not signing these vulnerable versions within relative short time intervals (unless they are hardware-based vulnerabilities), meaning that, contrary to Android, you can't downgrade iOS version with one exception explained below. Therefore, jailbreaking has become a very time sensitive procedure. Naturally, this creates a problem, when there is a major bump in iOS version (e.g. from 9 to 10) and there is no public jailbreak for the new OS.
```
Comment (Sjord) [inline Document/0x06b-Basic-Security-Testing.md]: > Jailbreaking iOS devices ... explained below.

I think this sentence is too long. Furthermore, it contains several subsentences between parenthesis or comma's, which make it hard to read.

File: Document/0x06b-Basic-Security-Testing.md [hunk 2]
```diff
@@ -27,36 +27,63 @@ If you want to get serious with iOS security testing, you need a Mac, for the si
...
+The iOS upgrade process is performed online and is based on challenge-response process. The device will perform OS installation if and only if the response to challenge is signed by Apple. This is what researchers call 'signing window' and explains the fact that you can't simply store the OTA firmware package downloaded via iTunes and load it to the device at any time. During minor iOS upgrades, it is possible that two versions are signed at the same time by Apple. This is the only case when you can possibly downgrade iOS version. You can check current signing window and download OTA Firmwares from the [IPSW Downloads website](https://ipsw.me).
```
Comment (Sjord) [inline Document/0x06b-Basic-Security-Testing.md]: > if and only if 

I think just "if" is clear enough.

**Comment by cpholguera** [comment]
Done, thanks @Sjord for your comments.

**Comment by sushi2k** [comment]
Thanks @cpholguera and @Sjord. I will merge it now.

**Pull request merged**